src/pipeline_edsl/to_workflow.ml

```diff
@@ -255,41 +253,116 @@ module Defaults = struct
   let results_dir = None
 end
 
+module Provenance_description = struct
+
+  type t = {
+    name: string;
+    sub_tree_arguments: (string * t) list;
```

Comment:
`string` and `json` are the "contants" (like reference-build, configuration, etc.) and the sub-tree is the provenance of the arguments of the EDSL function, e.g the input VCF here:

```ocaml
  let vcf_annotate_polyphen vcf =
    let v = get_vcf (AF.get_file vcf) in
    let output_vcf = (Filename.chop_extension v#product#path) ^ "_polyphen.vcf" in
    let reference_build = v#product#reference_build in
    Vcf (
      Tools.Vcfannotatepolyphen.run ~run_with ~reference_build ~vcf:v ~output_vcf
    )
    |> AF.with_provenance "vcf-annotate-polyphen" ["vcf", AF.get_provenance vcf]
```